tardis/gui/datahandler.py

```diff
@@ -0,0 +1,562 @@
+        Parameters
+        ----------
+            data: list of string
+                The data that is intended to be stored on the node.
+            parent: Node 
+                Another node which is the parent of this node. The 
+                root node has parent set to None.
+
+        Note
+        ----
+            A leaf node is a node that is the only child of its parent. 
+            For this tree this will always be the case. This is because
+            the tree stores the key at every node except the leaf node
+            where it stores the value for the key. So if in the dictionary
+            the value of a key is another dictionary, then it will 
+            be a node with no leafs. If the key has a value that is a 
+            value or a list then it will have one child that is a leaf.
+            The leaf can have no children. For example-
+
+            'a':val1
+            'b':{'c':val2
+                 'd':val3
+                 'e':{'f':val4
+                      'g':val5
+                         :val6
+                         :val7}} *In this case the key g can take values
+                                  val7, val5 and val6 and is currently
+                                  set to val5.
+
+            In the tree shown above all quoted values are keys in the 
+            dictionary and are non-leaf nodes in the tree. All values 
+            of the form valx are leaf nodes and are not dictionaries 
+            themselves. If the keys have non-dictionary values then they
+            have a leaf attached. And no leaf can have a child.
+        
+        """
+        self.parent = parent
+        self.children = []
+        self.data = data
+        self.siblings = {}  #For 'type' fields. Will store the nodes to 
+                            #enable disable on selection
+
+    def appendChild(self, child):
+        """Add a child to this node."""
+        self.children.append(child)
+        child.parent = self
+
+    def getChild(self, i):
+        """Get the ith child of this node. 
+
+        No error is raised if the cild requested doesn't exist. A 
+        None is returned in such cases.
+
+        """
+        if i < self.numChildren():
+            return self.children[i]
+        else:
+            return None
+
+    def numChildren(self):
+        """Number of children this node has."""
+        return len(self.children)
+
+    def numColumns(self):
+        """Returns the number of strings stored in the data attribute."""
+        return len(self.data)
+
+    def getData(self, i):
+        """Returns the ith string from the data list.
+
+        No error is raised if the data list index is exceeded. A None is
+        returned in such cases.
+
+        """
+        try:
+            return self.data[i]
+        except IndexError:
+            return None
+
+    def getParent(self):
+        """Return the parent of this node."""
+        return self.parent
+
+    def getIndexOfSelf(self):
```

Comment:
Hi Wolfgang,

I'll make the changes. I'm sorry I'm just a little busy now. I'll try to
finish up this PR and prepare a demo of the new layout by the weekend.

Yours sincerely,
Gaurav Gautam
On Jun 15, 2015 10:52 PM, "Wolfgang Kerzendorf" <notifications@github.com>
wrote:

> In tardis/gui/datahandler.py
> <https://github.com/tardis-sn/tardis/pull/324#discussion_r32444121>:
>
> > +        """Returns the ith string from the data list.
> > +
> > +        No error is raised if the data list index is exceeded. A None is
> > +        returned in such cases.
> > +
> > +        """
> > +        try:
> > +            return self.data[i]
> > +        except IndexError:
> > +            return None
> > +
> > +    def getParent(self):
> > +        """Return the parent of this node."""
> > +        return self.parent
> > +
> > +    def getIndexOfSelf(self):
>
> same here get_index_of_self
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/tardis-sn/tardis/pull/324/files#r32444121>.
>